microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshots.md

```diff
@@ -186,74 +191,41 @@ The **Containment** view is essentially a "bird's eye view" of the objects struc
-    };
-    return lC;
-}
-```
+![The Retainers section](./heap-snapshots-images/retainers-section.png)
 
-#### Demo webpage: Example 7: Eval is evil
+###### Hide cycles
 
-To analyze the impact of closures on memory, try out this example: open the demo webpage [Example 7: Eval is evil](https://microsoftedge.github.io/Demos/devtools-memory-heap-snapshot/example-07.html) in a new window or tab.
+In the **Retainers** section, when you analyze the objects which retain the selected object, you might encounter _cycles_. Cycles occur when the same object appears more than once in the retainer path of the selected object. In the **Retainers** section, a cycled object is indicated by being grayed out.
 
-<!-- You can view the source files for the Heap Snapshots demo pages in the [MicrosoftEdge/Demos > devtools-memory-heap-snapshot](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-memory-heap-snapshot) repo folder. -->
+To help simplify the retainer path, hide cycles in the **Retainers** section by clicking the **Filter edges** dropdown menu and then selecting **Hide cycled**:
 
-#### Demo webpage: Example 8: Recording heap allocations
+![The Filter edges dropdown menu in the Retainers section, 'Hide cycled' is selected](./heap-snapshots-images/filters-retainers-memory-tool-no-hide-cycled.png)
 
-You may also be interested in following up the above demo with this example that takes you through recording heap allocations: open the demo webpage [Example 8: Recording heap allocations](https://microsoftedge.github.io/Demos/devtools-memory-heap-snapshot/example-08.html) in a new window or tab.
+###### Hide internal nodes
 
-<!-- You can view the source files for the Heap Snapshots demo pages in the [MicrosoftEdge/Demos > devtools-memory-heap-snapshot](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-memory-heap-snapshot) repo folder. -->
+_Internal nodes_ are objects that are specific to V8 (the JavaScript engine in Microsoft Edge).
 
-<!--
-![Name functions to distinguish between closures](../media/memory-problems-domleaks.png)
--->
+To hide internal nodes from the **Retainers** section, in the **Filter edges** dropdown menu, select **Hide internal**.
 
 
 <!-- ====================================================================== -->
-## Filter a heap snapshot by node type
+## Filter heap snapshots by node types
 
-Use filters to focus on specific parts of a heap snapshot.  You can filter by node type if, for example, you're only interested in the strings or arrays from the heap.
+Use filters to focus on specific parts of a heap snapshot.
 
 When looking at all the objects in a heap snapshot in the **Memory** tool, it can be difficult to focus on specific objects or retaining paths.  Use the **Node Types** filter when looking at a heap snapshot, to focus on only specific types of nodes.  For example, to see only the arrays and string objects that are in the heap, select the **Array** and **String** entries in the **Node Types** filter.
```

Comment:
```suggestion
When looking at all the objects in a heap snapshot in the **Memory** tool, it can be difficult to focus on specific objects or retaining paths.  When looking at a heap snapshot, to focus on only specific types of nodes, use the **Node Types** filter, in the upper right.  For example, to see only the arrays and string objects in the heap, select the **Array** and **String** entries in the **Node Types** filter:
```

Reply:
does tooltip say **Node Types**?
ans: __